sympy/functions/special/delta_functions.py

```diff
@@ -499,8 +539,13 @@ def _eval_rewrite_as_sign(self, arg):
         sign
 
         """
+        if len(args) == 2:
+            arg, H0 = args[0], args[1]
+        else:
+            arg, H0 = args[0], None
         if arg.is_real:
-            return (sign(arg)+1)/2
+            if H0 == None or H0 == S.Half:
+                return (sign(arg)+1)/2
```

Comment:
This was my choice.  I expect that in simple usage, `Heaviside(0)` doesn't matter, and 
we can trust the user to mean it when they say they want `Heaviside` to be `sign`, and 
that `sign(0)` should override.

How about ...
```
        The value of the second argument of Heaviside must specify Heaviside(0)
        = 1/2 for rewritting as sign to be strictly equivalent.  For easier
        usage, we also allow this rewriting when Heaviside(0) is undefined.
```